packages/atlas/src/components/_ypp/YppAuthorizationStepModal/YppAuthorizationStepModal.tsx

```diff
@@ -0,0 +1,159 @@
+import { FC, useMemo } from 'react'
+
+import { BasicChannelFieldsFragment } from '@/api/queries/__generated__/fragments.generated'
+import appScreenshot from '@/assets/images/ypp-authorization/app-screenshot.webp'
+import { Text } from '@/components/Text'
+import { Button } from '@/components/_buttons/Button'
+import { SvgLogoYoutube } from '@/components/_icons'
+import { DialogModal, DialogModalProps } from '@/components/_overlays/DialogModal'
+
+import {
+  AdditionalSubtitle,
+  Content,
+  HeaderIconsWrapper,
+  Img,
+  StyledSvgAtlasLogoShort,
+  StyledSvgControlsConnect,
+} from './YppAuthorizationStepModal.styles'
+
+import { DetailsFormDialog } from '../DetailsFormDialog'
+import { RequirementsDialog } from '../RequirementsDialog'
+import { SelectChannelDialog } from '../SelectChannelDialog'
+import { TermsAndConditionsDialog } from '../TermsAndConditionsDialog'
+
+type Step = 'select-channel' | 'requirements' | 'details' | 'terms-and-conditions' | 'summary' | 'connect-with-yt'
+
+export type YppAuthorizationStepModalProps = {
+  onPrimaryClick?: () => void
+  onBackClick?: () => void
+  onCancelClick?: () => void
+  step: Step
+  channels: BasicChannelFieldsFragment[]
+} & Pick<DialogModalProps, 'show'>
+
+export const YppAuthorizationStepModal: FC<YppAuthorizationStepModalProps> = ({
+  show,
+  onBackClick,
+  onCancelClick,
+  channels,
+  step,
+}) => {
+  const isSummary = step === 'summary'
+  const isConnectWithYt = step === 'connect-with-yt'
+  const isSelectChannel = step === 'select-channel'
+  const authorizationStep = useMemo(() => {
+    switch (step) {
+      case 'select-channel':
+        return {
+          title: 'Select channel',
+          description: 'Select the Atlas channel you want your YouTube channel to be connected with.',
+          primaryButton: { text: 'Select channel' },
+          component: <SelectChannelDialog channels={channels} />,
+        }
+      case 'requirements':
+        return {
+          title: 'Requirements',
+          description:
+            'Before you can apply to the program, make sure both your Atlas and YouTube channels meet the below conditions.',
+          primaryButton: {
+            text: 'Authorize with YouTube',
+          },
+          component: <RequirementsDialog />,
+        }
+      case 'details':
+        return {
+          title: 'Details',
+          description: 'We need your email address to send you payment information. No spam or marketing materials.',
+          primaryButton: {
+            text: 'Continue',
+          },
+          component: <DetailsFormDialog />,
+        }
+      case 'terms-and-conditions':
+        return {
+          title: 'Terms & conditions',
+          description:
+            'Once automatic YouTube videos sync is available, in order for it to work, your Atlas channel [NEEDS TO DO WHAT?]. This is purely a technical measure and does not affect ownership and rights to the content uploaded to you Atlas channel.',
```

Comment:
NEEDS TO DO WHAT? 
![image](https://user-images.githubusercontent.com/51168865/190598126-c65a10fd-1d3f-473b-a190-bea70a0829a8.png)

Let's add a todo to fix this copy, or replace it with some proper text.